packages/zoe/src/zoe.js

```diff
@@ -696,29 +958,47 @@ const makeZoe = (additionalEndowments = {}) => {
                     completeOffers(instanceHandle, harden([offerHandle])),
                 }),
               );
-              // Add an object with a cancel method to redemptionResult in
+              // Add an object with a cancel method to offerResult in
               // order to cancel on demand.
             } else if (exitKind === 'onDemand') {
-              redemptionResult.cancelObj = {
+              offerResult.cancelObj = {
                 cancel: () =>
                   completeOffers(instanceHandle, harden([offerHandle])),
               };
             } else {
               assert(
                 exitKind === 'waived',
-                details`exit kind was not recognized: ${exitKind}`,
+                details`exit kind was not recognized: ${openDetail(exitKind)}`,
               );
             }
 
             // if the exitRule.kind is 'waived' the user has no
             // possibility of cancelling
-            return harden(redemptionResult);
+            return harden(offerResult);
           };
           return Promise.all(paymentDepositedPs)
             .then(recordOffer)
-            .then(makeRedemptionResult);
+            .then(makeOfferResult);
         });
       },
+
+      // TODO(msm): remove this deprecated one in this PR
```

Comment:
Agreed, that sounds good.

Reply:
That comment was hopeful ;)

As explained in previous comments, I wonder if we can remove it in the next PR

Reply:
Resolving as a dup of above conversation